src/coreclr/jit/gentree.cpp

```diff
@@ -30122,7 +30207,8 @@ NamedIntrinsic GenTreeHWIntrinsic::GetHWIntrinsicIdForBinOp(Compiler*  comp,
         case GT_DIV:
         {
 #if defined(TARGET_XARCH)
-            assert(varTypeIsFloating(simdBaseType) || varTypeIsInt(simdBaseType));
+            assert(varTypeIsFloating(simdBaseType) ||
+                   (varTypeIsIntegral(simdBaseType) && !varTypeIsLong(simdBaseType)));
```

Comment:
We can only have floating or integral node types, so this can probably be simplified to:
```
```suggestion
            assert(varTypeIsFloating(simdBaseType) || !varTypeIsLong(simdBaseType));
```